cli/tsc_config.rs

```diff
@@ -167,6 +169,42 @@ pub fn parse_raw_config(config_text: &str) -> Result<Value, AnyError> {
   Ok(jsonc_to_serde(jsonc))
 }
 
+/// Parse provided options and return error if config contains
+/// ignored options.
+pub fn parse_config_with_ignored_options(
```

Comment:
Why aren't we just using/modifying `TsConfig::merge_user_config`?  Also `IgnoredCompilerOptions` already includes a meaningful error message that duplicates what we are adding here.

I think it is better to just return what was ignored to the API and let the API figure out if that is a meaningful and what to do with it.

Reply:
@kitsonk because we have a JSON string with config and not a path - `merge_user_config` is only useful if you want to load and merge config from a file.

Reply:
> I think it is better to just return what was ignored to the API and let the API figure out if that is a meaningful and what to do with it.

So change return type to return list of ignored options?

Reply:
🤦 oh yeah regarding the `merge_user_config`... Maybe we should have `merge_tsconfig(path)` to replace what we already do and make `merge_user_config(Value)` for what you need to do here?